pandas/io/parsers.py

```diff
@@ -1176,13 +1176,18 @@ def _should_parse_dates(self, i):
         if isinstance(self.parse_dates, bool):
             return self.parse_dates
         else:
-            name = self.index_names[i]
+            if self.index_names is not None:
+                name = self.index_names[i]
```

Comment:
right. yep must not have taken that path at all, e.g. ``read_csv(..., parse_dates=['column_name'], index_col=0)`` is all that this is doing.......

if you want to look and see maybe missing something obvious, would be great.